### ISSUE TITLE
Prune fails: "Integrity check failed: Data seems to be missing."

### ISSUE THREAD

**Issue by Mebus** [restic/restic#3766]
Hallo folks,

I want to prune a repo, but I get this:

```
 restic prune --cache-dir /var/restic-cache  -r mybackup/       
 enter password for repository: 
repository 8166a112 opened successfully, password is correct
loading indexes...
loading all snapshots...
finding data that is still in use for 28 snapshots
[0:57] 100.00%  28 / 28 snapshots...
searching used packs...
{<data/23060307> <data/2ad8241b> <data/3a014022> <data/465e48c3> <data/4e2eb3be> <data/5e938c0c> <data/6003057d> <data/61ec201b> <data/66f56c46> <data/84dabd92> <data/8772df38> <data/908fa665> <data/942ea0e4> <data/b1b82d07> <data/b3bb5f62> <data/b466d871> <data/b4d06e3b> <data/be7dc879> <data/c602e1ae> <data/c71f5ad4> <data/dc672d00> <data/fc081bee>} not found in the index

Integrity check failed: Data seems to be missing.
Will not start prune to prevent (additional) data loss!
Please report this error (along with the output of the 'prune' run) at
https://github.com/restic/restic/issues/new/choose
Fatal: index is not complete
```

I already ran `recover` and `rebuild-index` but that did not help. What can I do to fix this?

Mebus

**Comment by MichaelEischer** [comment]
Please include the information requested in https://raw.githubusercontent.com/restic/restic/master/.github/ISSUE_TEMPLATE/Bug.md .


Which restic version are you using? Have you upgraded restic recently? Did you notice any errors reported in prior restic runs or did something unexpected happen? Maybe an interrupted network connection or did the host go into standby while restic was running? Is this the first failing `prune` run? Do you have logs of the `backup` and `prune` run before the failing one? Did restic report any errors there?

How often do you run `backup`, `check` and/or `prune`?

Which backend are you using?

Please run `restic check --read-data` to analyze the damage to the repository.

Which output did `rebuild-index` generate?

It would be helpful to know which snapshots are missing blobs: Please run `restic find --blob 23060307 2ad8241b 3a014022 465e48c3 4e2eb3be 5e938c0c 6003057d 61ec201b 66f56c46 84dabd92 8772df38 908fa665 942ea0e4 b1b82d07 b3bb5f62 b466d871 b4d06e3b be7dc879 c602e1ae c71f5ad4 dc672d00 fc081bee` to find the affected snapshots.

**Comment by rmoesbergen** [comment]
Hi, I'm having the same issue:
```
repository e47b83f6 opened successfully, password is correct
loading indexes...
loading all snapshots...
finding data that is still in use for 10 snapshots
[0:01] 100.00%  10 / 10 snapshots...
searching used packs...
{<data/04ee0b37>} not found in the index

Integrity check failed: Data seems to be missing.
Will not start prune to prevent (additional) data loss!
Please report this error (along with the output of the 'prune' run) at
https://github.com/restic/restic/issues/new/choose
Fatal: index is not complete
```

restic find --blob 04ee0b37 outputs:
```
repository e47b83f6 opened successfully, password is correct
Found blob 04ee0b374c0ce16479ed26e01bfa05238c7a316ca6fd3ff3369437082dacc12d
 ... in file /export/backups/mail-20220701-230001.tgz
     (tree 8b5d6afc2b4581c31511a147ae5ba5775193e4f4a714edc1b88f06bc07ddcb93)
 ... in snapshot 5dc6820d (2022-07-01 23:01:08)
Found blob 04ee0b374c0ce16479ed26e01bfa05238c7a316ca6fd3ff3369437082dacc12d
 ... in file /export/backups/mail-20220701-230001.tgz
     (tree 2836db3a0f9507fd3d2e0394f6052896716df246e1f380d1689a8244afd3016f)
 ... in snapshot 3cf7707d (2022-07-02 23:01:07)
```

This is the second machine that I encouter this issue on, seems there's a bug somewhere. I'm using rclone with 'pCloud' backend as the backup destination on both counts. How can I 'repair' the repository? I don't mind losing the 2 snapshots above.

**Comment by rmoesbergen** [comment]
- Which restic version are you using?
`restic 0.13.1 compiled with go1.18 on linux/amd64`
`rclone v1.58.1`, using the 'pcloud' backend.

- Have you upgraded restic recently?
Yes.

- Did you notice any errors reported in prior restic runs or did something unexpected happen?
I only keep logs for the latest run, so can't say

- Is this the first failing prune run?
Probably not. Also not the first VM that I've seen this on (same restic/rclone versions, same backend)

- Do you have logs of the backup and prune run before the failing one?
No, sorry

- How often do you run backup, check and/or prune?
Daily:  first backup, then 'forget', then 'prune'

The output in my first comment is the ouput of `restic check`. After running rebuild-index and removing the 2 damaged snapshots _and_ doing another backup, the repository is now OK again:

```
using temporary cache in /tmp/restic-check-cache-3718045511
repository e47b83f6 opened successfully, password is correct
created new cache in /tmp/restic-check-cache-3718045511
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
[0:05] 100.00%  9 / 9 snapshots...
no errors were found
```

Will keep an eye out to see if this happens again.

**Comment by MichaelEischer** [comment]
> * Daily:  first backup, then 'forget', then 'prune'

That's quite a high frequency to run `prune` with, but it is helpful insofar as it suggests, that the prune run before the failing one was probably the culprit.


> This is the second machine that I encouter this issue on, seems there's a bug somewhere.

My impression at the moment is that there are primarily two categories of integrity problems detected during prune: either a single blob is missing or quite a bunch of blobs. The latter is probably caused by some data file in the repository disappearing, whereas the former might be related to bitflips in hardware. Or maybe there's a bug lurking somewhere. But the number of reported prune problems is low enough to not provide any useful leads so far.

**Comment by MichaelEischer** [comment]
Prune has gained a few additional sanity checks in #3899 (not yet released )and #3829 (included in 0.14.0) which might catch the problem reported here.

I'm closing this issue as there is nothing left to investigate here.
